source/connect/tls.txt

```diff
@@ -0,0 +1,245 @@
+        the API Documentation for `Mongo::Client <{+api-root+}/Mongo/Client.html>`__ .  
+
+        In the following example, the TLS certificate and corresponding private key 
+        are provided in separate files:
+
+        .. literalinclude:: /includes/connect/tls.rb
+            :language: ruby
+            :dedent:
+            :start-after: start-enable-tls-settings
+            :end-before: end-enable-tls-settings
+        
+        The TLS certificate and private key can be contained in the same file, but 
+        both the certificate and private key options must still be specified. In the 
+        following example, the TLS certificate and the private key are both defined 
+        in the same ``client.pem`` file:
+
+        .. literalinclude:: /includes/connect/tls.rb
+            :language: ruby
+            :dedent:
+            :start-after: start-enable-tls-settings-same-file
+            :end-before: end-enable-tls-settings-same-file
+
+    .. tab:: Connection String
+        :tabid: uri
+
+        To configure certificates, you must specify the following URI options:
+        
+        - ``tlsCertificateKeyFile``: The certificate and key file used to identify
+          identify the connection against MongoDB. Both the certificate and key
+          must be stored in the same file provided to this option.
+        - ``tlsCAFile``: The file containing the concatenated certificate authority (CA)
+          certificates used to validate certificates passed from the other end of the 
+          connection. The driver will use the default system root certificate 
+          store as the trust anchor, if the option is not designated. 
+
+        .. literalinclude:: /includes/connect/tls.rb
+            :language: ruby
+            :dedent:
+            :start-after: start-enable-tls-uri
+            :end-before: end-enable-tls-uri
+        
+        The file containing the certificate and key usually has a``.crt`` or ``.pem`` 
+        extension.
+
+        URI option values must be properly URI escaped. This applies, for example, 
+        to slashes (/) in the paths, which are encoded as ``%2f``.
+
+Modify the SSL Context
+----------------------
+
+If your TLS configuration requires customization, you can set TLS context hooks
+by adding a native Ruby class ``Proc`` object to the ``Mongo.tls_context_hooks`` 
+array. The ``Proc`` should be added to the array before creating any ``Mongo::Client`` 
+instances. 
+
+For example, the following code could be placed in an initializer to enable the 
+AES256-SHA cipher as the only cipher to be used for TLS:
+
+.. literalinclude:: /includes/connect/tls.rb
+    :language: ruby
+    :dedent:
+    :start-after: start-modify-context
+    :end-before: end-modify-context
+
+To learn more about the SSL context options available, see the Ruby
+documentation for 
+`SSLContext <https://ruby-doc.org/stdlib-2.7.0/libdoc/openssl/rdoc/OpenSSL/SSL/SSLContext.html>`__.
```

Comment:
nit:

```suggestion
`SSLContext. <https://ruby-doc.org/stdlib-2.7.0/libdoc/openssl/rdoc/OpenSSL/SSL/SSLContext.html>`__
```

Reply:
Same as above, I like to put periods outside the hyperlink since it's not a part of the context of the actual link! Please let me know if there is a documented standard, though

Reply:
there's nothing in the style guide--just what i've seen as recommended. either way works though